pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -86,10 +87,20 @@ type Role string
 
 // K8SSelectorConfig is Kubernetes Selector Config
 type K8SSelectorConfig struct {
-	// +kubebuilder:validation:Required
-	Role  Role   `json:"role"`
-	Label string `json:"label,omitempty"`
-	Field string `json:"field,omitempty"`
+	// Role specifies the type of Kubernetes resource to limit the service discovery to.
+	// Accepted values are: Node, Pod, Endpoints, EndpointSlice, Service, Ingress.
+	// +required
```

Comment:
Hmm not sure why we didn't add enum validator in the first place. @slashpai do you remember?